src/nav/AppNavigator.js

```diff
@@ -80,66 +84,98 @@ export type AppNavigatorParamList = {|
 
 export type AppNavigationProp<
   RouteName: $Keys<AppNavigatorParamList> = $Keys<AppNavigatorParamList>,
-> = NavigationStackProp<{|
-  ...NavigationStateRoute,
-  params: $ElementType<GlobalParamList, RouteName>,
+> = StackNavigationProp<GlobalParamList, RouteName>;
+
+export type AppNavigationRouteProp<
+  RouteName: $Keys<AppNavigatorParamList> = $Keys<AppNavigatorParamList>,
+> = RouteProp<GlobalParamList, RouteName>;
+
+const Stack = createStackNavigator<GlobalParamList, AppNavigatorParamList, AppNavigationProp<>>();
+
+type SelectorProps = $ReadOnly<{|
+  hasAuth: boolean,
+  accounts: Account[],
+  haveServerData: boolean,
```

Comment:
I think this means that the caller, ZulipNavigationContainer, can stop getting these three from Redux. It'd be good to pair the deletion with the addition, so that the diff is visibly just moving the logic.

I think this is also logically more or less independent of the change in the nav API, so it'd simplify things to make it a separate diff. (This commit in particular has a large quantity of boring mechanical stuff happening, so it'd be good to keep more-interesting changes out of it.) I think it could pretty well happen either before or after this one -- either `createAppNavigator` starts taking these instead of its old arguments and calling `getInitialRouteInfo` itself, or `AppNavigator` at first still takes the old arguments and later starts connecting to Redux for itself.

Reply:
> or `AppNavigator` at first still takes the old arguments and later starts connecting to Redux for itself.

OK! I've taken this approach, with a new commit at the tip.